include/AdePT/kernels/gammas.cuh

```diff
@@ -89,6 +90,16 @@ __global__ void TransportGammas(adept::TrackManager<Track> *gammas, Secondaries
       continue;
     }
 
+#ifdef ASYNC_MODE
+    if (InFlightStats->perEventInFlightPrevious[currentTrack.threadId] < allowFinishOffEvent[currentTrack.threadId] &&
+        InFlightStats->perEventInFlightPrevious[currentTrack.threadId] != 0) {
+      printf("Thread %d Killing gammas when killing the %d last particles of event %d E=%f lvol=%d after %d steps.\n",
```

Comment:
This is, I guess, temporary, until we manage to send those particles back to the CPU